### PR TITLE
Banish nulls

### PR DESCRIPTION
**Pull request by buntec**

### CODE REVIEW

File: snabbdom/src/main/scala/snabbdom/VNode.scala [hunk 1]
```diff
@@ -62,13 +62,17 @@ class VNode private (
 
 object VNode {
 
+  val empty =
+    new VNode(None, VNodeData.empty, None, None, Some(""), None, None)
```
Comment (armanbilge) [inline snabbdom/src/main/scala/snabbdom/VNode.scala:66]: Isn't this global mutable state?

Comment (buntec) [reply snabbdom/src/main/scala/snabbdom/VNode.scala:66]: Oops, should be a `def`

Comment (armanbilge) [reply snabbdom/src/main/scala/snabbdom/VNode.scala]: I wonder if we should make it `empty()` to distinguish it as allocating mutable state 🤔

File: snabbdom/src/main/scala/snabbdom/init.scala [hunk 2]
```diff
@@ -238,78 +232,69 @@ object init {
...
 
-    def getOrNull(vnodes: Array[VNode], idx: Int): VNode = {
-      if (idx >= 0 && idx < vnodes.length) {
-        vnodes(idx)
-      } else {
-        null
-      }
-    }
-
     def updateChildren(
```
Comment (armanbilge) [inline snabbdom/src/main/scala/snabbdom/init.scala:251]: Nice, so you reworked this!

**Review by armanbilge — APPROVED** [review]
Admittedly I didn't think too hard about these changes, but seeing the tests pass without a single change is usually a good sign 😉

**Pull request merged**